provider/cmd/app/app.go

```diff
@@ -125,22 +127,27 @@ func (o *Options) MarkFlagsRequired(cmd *cobra.Command) {
 	_ = cmd.MarkFlagRequired("supported-machine-classes")
 }
 
-func Command() *cobra.Command {
+// Command creates cobra command
+// It return pointer to pointer for zap logger because flags and logger are initialized in Execute function
+func Command() (*cobra.Command, **originzap.Logger) {
 	var (
-		zapOpts = zap.Options{Development: true}
+		zapOpts = zap.Options{Development: false}
 		opts    Options
 	)
 
+	var logger *originzap.Logger
 	cmd := &cobra.Command{
 		Use: "libvirt-provider",
 		PersistentPreRun: func(cmd *cobra.Command, args []string) {
-			logger := zap.New(zap.UseFlagOptions(&zapOpts))
-			ctrl.SetLogger(logger)
+			logger = zap.NewRaw(zap.UseFlagOptions(&zapOpts))
```

Comment:
What is the motivation behind using the Raw logger here?

Reply:
flushing of buffers

Reply:
Maybe instead of using `NewRaw` we should look at those (took if from the code docs):

```
// New constructs a new Logger from the provided zapcore.Core and Options. If
// the passed zapcore.Core is nil, it falls back to using a no-op
// implementation.
//
// This is the most flexible way to construct a Logger, but also the most
// verbose. For typical use cases, the highly-opinionated presets
// (NewProduction, NewDevelopment, and NewExample) or the Config struct are
// more convenient.
```

Reply:
zap library isn't real zap library but it is controller-runtime zap library which add abstraction for zap. Primary differences is support of flags which are bind later. This isn't possible do with origin zap without define own set of flags. https://github.com/kubernetes-sigs/controller-runtime/blob/v0.16.3/pkg/log/zap/zap.go#L229